src/Providers/Tenants/RouteProvider.php

```diff
@@ -17,31 +17,33 @@
 use Hyn\Tenancy\Contracts\CurrentHostname;
 use Hyn\Tenancy\Models\Hostname;
 use Illuminate\Config\Repository;
-use Illuminate\Foundation\Support\Providers\RouteServiceProvider;
 use Illuminate\Routing\RouteCollection;
 use Illuminate\Routing\Router;
+use Illuminate\Support\ServiceProvider;
 
-class RouteProvider extends RouteServiceProvider
+class RouteProvider extends ServiceProvider
 {
     public function boot()
     {
-        $this->loadRoutes();
-    }
-
-    public function map(Repository $config, Router $router)
-    {
+        /** @var Repository $config */
+        $config = $this->app->make(Repository::class);
         $path = $config->get('tenancy.routes.path');
 
-        if ($path && $config->get('tenancy.hostname.auto-identification')) {
+        /** @var Router $router */
+        $router = $this->app->make(Router::class);
+
+        if ($path) {
             /** @var Hostname $hostname */
             $hostname = $this->app->make(CurrentHostname::class);
 
             if ($hostname && file_exists($path)) {
-                if ($config->get('tenancy.routes.replace-global')) {
-                    $router->setRoutes(new RouteCollection());
-                }
+                $this->app->booted(function () use ($config, $router, $path) {
```

Comment:
Hm, interesting fix... I was looking for something similar, but hadn't put much effort into it.  Nice job!  I'll give it a test later tonight or tomorrow.

Reply:
@fletch3555 just out of transparency; I actually looked at both the Laravel Router and RouteServiceProvider (not the app one but the Illuminate one). It seems that the service provider adds a booted callback as well to refresh the routes (not to add, remove anything).